src/index.tsx

```diff
@@ -585,32 +587,40 @@ const ModalizeBase = (
   const renderContent = (): JSX.Element => {
     const keyboardDismissMode = isIos ? 'interactive' : 'on-drag';
 
+    const passedOnScrollBeginDrag = (flatListProps ?? sectionListProps ?? scrollViewProps)
+      ?.onScrollBeginDrag;
+
     const opts = {
       ref: contentView,
       bounces: enableBounces,
-      onScrollBeginDrag: Animated.event([{ nativeEvent: { contentOffset: { y: beginScrollY } } }], {
-        useNativeDriver: USE_NATIVE_DRIVER,
-      }),
+      onScrollBeginDrag: (e: NativeSyntheticEvent<NativeScrollEvent>): void => {
+        beginScrollY.setValue(e.nativeEvent.contentOffset.y);
```

Comment:
Could we keep the animated.event/useNativeDriver to have it running on the UI thread while setValue happen on the JS side?